pymeasure/instruments/anaheimautomation/dpseriesmotorcontroller.py

```diff
@@ -0,0 +1,377 @@
+            read_termination="\r",
+            timeout=2000, 
+            **kwargs
+        )
+        
+        if isinstance(self.adapter, VISAAdapter):
+            self.adapter.connection.baud_rate = 38400
+    
+    @property
+    def encoder_enabled(self):
+        """
+        Return the value of the _encoder_enabled flag.
+        """
+        return self._encoder_enabled
+
+    @encoder_enabled.setter
+    def encoder_enabled(self, en):
+        """ Set the value of the _encoder_enabled flag. When set, asynchronous coroutine methods like step_async() will
+            query the "encoder_position" property instead of the "position" property.
+
+        :param en: (bool) boolean value to set the _encoder_enabled flag with
+        """
+        self._encoder_enabled = bool(en)
+    
+    @property
+    def step_position(self):
+        """
+        Return the value of the motor position measured in steps counted by the motor controller or, if encoder_enabled
+        is set, return the steps counted by an externally connected encoder.
+        """
+        if self._encoder_enabled:
+            pos = self.ask("VEP")
+        else:
+            pos = self.ask("VZ")    
+        return int(pos)
+    
+    @step_position.setter
+    def step_position(self, pos):
+        """ Sends command to the motor controller to move to the desired step position. Note that in DP series
+            controller instrument manuals, this property corresponds to the 'absolute position' command. In this driver,
+            the `absolute_position` property implements a conversion between steps to absolute units whereas the DP
+            series manuals refer to `absolute position` as measured in steps.
+        """
+        step_pos = int(pos)
+        if -8388607 < step_pos < 8388607:
+            self.write("P%i" % step_pos)
+            self.go()
+        else:
+            raise ValueError("Provided step position out of valid range!")
+    
+    @property
+    def absolute_position(self):
+        """
+        Return the absolute position of the motor in units determined by the steps_to_position() method.
+        """
+        step_pos = self.step_position
+        return self.steps_to_absolute(step_pos)
+    
+    @absolute_position.setter
+    def absolute_position(self, abs_pos):
+        """ 
+        Move the motor to the absolute position provided. Note that the absolute_to_steps()
+        method must be implemented in a subclass for this to work.
+        
+        :param abs_pos: absolute position in units determined by the absolute_to_steps() method.
+        """
+        steps_pos = self.absolute_to_steps(abs_pos)
+        self.step_position = steps_pos
+    
+    def absolute_to_steps(self, pos):
+        """ Convert an absolute position to a number of steps to move. This must be implemented in subclasses.
+        
+        :param pos: Absolute position in the units determined by the subclassed position_to_steps() method.
```

Comment:
Still old method name, from a renaming I guess.

Reply:
otherwise this approach of punting implementation to a subclass is fine by me, it will always depend on the physical setup by the user anyway, and this way they can implement more complex transformations than a simple multiplication-by-gear-ratio, e.g. if the motor drives a cam or whatnot.